vlingo-reactive-messaging-patterns/src/main/java/io/vlingo/reactive/messaging/patterns/recipientlist/MountainSuppliesOrderProcessor.java

```diff
@@ -0,0 +1,296 @@
+    
+    public MountainSuppliesOrderProcessor( TestUntil until, TestUntil untilRegistered )
+    {
+        this.until = until;
+        this.untilRegistered = untilRegistered;
+        this.interestRegistry = new HashMap<>();
+    }
+
+    /* @see io.vlingo.reactive.messaging.patterns.recipientlist.OrderProcessor#registerPriceQuoteInterest(io.vlingo.reactive.messaging.patterns.recipientlist.PriceQuoteInterest) */
+    @Override
+    public void registerPriceQuoteInterest( PriceQuoteInterest interest )
+    {
+        logger().log( String.format( "%s interested", interest.path ));
+        interestRegistry.put( interest.path, interest );
+        untilRegistered.happened();
+    }
+
+    /* @see io.vlingo.reactive.messaging.patterns.recipientlist.OrderProcessor#requestForQuote(io.vlingo.reactive.messaging.patterns.recipientlist.RetailBasket) */
+    @Override
+    public void requestForQuote( RetailBasket basket )
+    {
+        List<QuoteProcessor> recipientList = calculateRecipientList( basket );
+        dispatchTo( basket, recipientList );
+    }
+
+    /* @see io.vlingo.reactive.messaging.patterns.recipientlist.OrderProcessor#remittedPriceQuote(io.vlingo.reactive.messaging.patterns.recipientlist.PriceQuote) */
+    @Override
+    public void remittedPriceQuote( PriceQuote quote )
+    {
+        logger().log( String.format( "OrderProcessor received price quote: %s", quote ));
+        until.happened();
+    }
+
+    protected List<QuoteProcessor> calculateRecipientList( RetailBasket basket )
+    {
+        List<QuoteProcessor> recipientList = new ArrayList<>();
+        for ( PriceQuoteInterest interest : interestRegistry.values() )
+        {
+            if (( interest.lowTotalRetail <= basket.totalRetailPrice )
+             && ( interest.highTotalRetail >= basket.totalRetailPrice ))
+            {
+                recipientList.add( interest.quoteProcessor );
+            }
+        }
+        return recipientList;
+    }
+
+    protected void dispatchTo( RetailBasket basket, List<QuoteProcessor> recipientList )
+    {
+        for ( QuoteProcessor quoteProcessor : recipientList )
+        {
+            for ( RetailItem item : basket.retailItems )
+            {
+                quoteProcessor.requestPriceQuote( basket.rfqId, item.itemId, item.retailPrice, basket.totalRetailPrice );
+            }
+        }
+    }
+
+    // QUOTE PROCESSORS
+    public static class BudgetHikersPriceQuotes 
+    extends Actor 
+    implements QuoteProcessor
+    {
+        public final OrderProcessor orderProcessor;
+        
+        public BudgetHikersPriceQuotes( final OrderProcessor aProcessor )
+        {
+            this.orderProcessor = aProcessor;
+        }
+
+        /* @see io.vlingo.actors.Actor#beforeStart() */
+        @Override
+        protected void beforeStart()
+        {
+            orderProcessor.registerPriceQuoteInterest( new PriceQuoteInterest( this.getClass().getName(), this, 1d, 1000d ));
```

Comment:
This is why I would change `path` to `type` in `PriceQuoteInterest`.

Also, `this` must be `selfAs(QuoteProcessor.class)`. Otherwise you will cause a direct method invocation rather than message send.

Reply:
Cool and geeky: selfAs rather than this for message send.

Reply:
Yeah, problem being that `this` works, and if you aren't conscious of what you are doing you can cause some really strange bugs. Imagine an actor is handling a true sent message and a method invocation simultaneously. Bye-bye protected state.